### PR TITLE
NAS-106093 / 13.0 / SES firmware update

### PR DESCRIPTION
**Pull request by themylogin**

### CODE REVIEW

File: src/freenas/usr/local/share/ses_firmware/manifest.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "file": "rainbow_4.3.0.5_CombinedImage.bin",
+  "version": "0430"
```
Comment (themylogin) [inline src/freenas/usr/local/share/ses_firmware/manifest.json:3]: @darkfiberiru is `rainbow_4.3.0.6_CombinedImage.bin` possible? What will be a string in `camcontrol devlist` then?

File: src/freenas/usr/local/sbin/firmware_update_ses.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        self.found = False
+
+        with open(os.path.join(UPDATE_PATH, "manifest.json")) as f:
+            manifest = json.load(f)
+
+        self.update_path = os.path.join(UPDATE_PATH, manifest["file"])
+        self.update_version = manifest["version"]
+
+        devlist = run(["camcontrol", "devlist"]).stdout
+        m = re.search(r"<CELESTIC R0904-F0001-01 ([0-9]+)>.*\((ses[0-9]+),", devlist)
```
Comment (yocalebo) [inline src/freenas/usr/local/sbin/firmware_update_ses.py:47]: What expansion shelf is this? Is the a newer one? The ES60 shelves show up in camcontrol as `CELESTIC R0904-F1001`.

Also, do we want to alert that "Enclosure not found" for systems that don't have this enclosure?

File: src/freenas/etc/ix.rc.d/ix-update [hunk 3]
```diff
@@ -104,7 +104,14 @@ db_update()
 			rm ${HBA_FIRMWARE_UPDATE}
 		fi
 	fi
-			
+
+	if [ -f $NEED_UPDATE_SENTINEL ]; then
```
Comment (william-gr) [inline src/freenas/etc/ix.rc.d/ix-update:108]: Isnt this only actually used for config upload? For actual upgrades it needs to be in build repo, right?

File: src/freenas/usr/local/sbin/firmware_update_ses.py (hunk 2 above)
Comment (themylogin) [reply src/freenas/usr/local/sbin/firmware_update_ses.py:47]: @darkfiberiru sorry, do you any news on this?

**Comment by themylogin** [comment]
backport

**Comment by themylogin** [comment]
No one needs this

**Pull request closed without merging**